### PR TITLE
fixes missing compatibility coverage for protocol 1.1 error codecs

### PR DESCRIPTION
**Pull request by mustafaiman**
ErrorCodecFileGenerator generates only newly added errors for each protocol version.
ErrorCodecCompatibilityTest used to look for all kinds of throwables in each protocol version which is not correct behaviour. This PR fixes it.
Adds `Nightly` category to test so it is run in daily builds.

### CODE REVIEW

**Comment by ihsandemir** [comment]
👍

**Comment by mustafaiman** [comment]
verify

**Comment by mustafaiman** [comment]
verify

**Comment by mustafaiman** [comment]
run-lab-run

**Comment by mustafaiman** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/client/protocol/compatibility/ErrorCodecCompatibilityTest.java [hunk 1]
```diff
@@ -38,22 +40,23 @@
 
 @RunWith(Parameterized.class)
 @Parameterized.UseParametersRunnerFactory(HazelcastParametersRunnerFactory.class)
+@Category(NightlyTest.class)
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/client/protocol/compatibility/ErrorCodecCompatibilityTest.java]: If this test is important for code coverage, then it has to be a `SlowTest` or `QuickTest`. If it's just for correctness, then `NightlyTest` is absolutely fine.

**Comment by Donnerbart** [comment]
:+1: (once the test category has been verified)

File: hazelcast/src/test/java/com/hazelcast/client/protocol/compatibility/ErrorCodecCompatibilityTest.java (hunk 1 above)
Comment (mustafaiman) [inline hazelcast/src/test/java/com/hazelcast/client/protocol/compatibility/ErrorCodecCompatibilityTest.java]: Aim is to keep compatibility in place, not increase coverage.

Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/client/protocol/compatibility/ErrorCodecCompatibilityTest.java]: Then the category is totally fine :)

**Pull request merged**